docs/developer-docs.rst

```diff
@@ -166,3 +166,181 @@ release is described here.
+``examples/plugin_example/`` and run ``pip install .`` This will add the
+``gwexample`` plugin to girder\_worker. If you then run girder\_worker
+with a log level of 'info' (e.g. ``girder-worker -l info``) you should
+see the following output:
+
+::
+
+    (girder)$ girder-worker -l info
+
+     -------------- celery@minastirith v3.1.23 (Cipater)
+    ---- **** -----
+    --- * ***  * -- Linux-4.8.6-1-ARCH-x86_64-with-glibc2.2.5
+    -- * - **** ---
+    - ** ---------- [config]
+    - ** ---------- .> app:         girder_worker:0x7f69bfff1050
+    - ** ---------- .> transport:   amqp://guest:**@localhost:5672//
+    - ** ---------- .> results:     amqp://
+    - *** --- * --- .> concurrency: 32 (prefork)
+    -- ******* ----
+    --- ***** ----- [queues]
+     -------------- .> celery           exchange=celery(direct) key=celery
+
+
+    [tasks]
+      . girder_worker.convert
+      . girder_worker.run
+      . girder_worker.validators
+      . gwexample.analyses.tasks.fibonacci
+
+    [2016-11-08 12:22:56,163: INFO/MainProcess] Connected to amqp://guest:**@127.0.0.1:5672//
+    [2016-11-08 12:22:56,184: INFO/MainProcess] mingle: searching for neighbors
+    [2016-11-08 12:22:57,198: INFO/MainProcess] mingle: all alone
+    [2016-11-08 12:22:57,218: WARNING/MainProcess] celery@minastirith ready.
+
+Notice that the task ``gwexample.analyses.tasks.fibonacci`` is now
+available. With the girder-worker worker running, you should be able to
+execute ``python example_client.py`` in the current working directory.
+After a brief delay, this should print out ``121393`` - the Fibonacci
+number for 26.
+
+Writing your own plugin
+-----------------------
+
+Adding additional tasks to the girder\_worker infrastructure is easy and
+takes three steps. (1) Creating tasks, (2) creating a plugin class and
+(3) adding a ``girder_worker_plugins`` entry point to your setup.py.
+
+Creating tasks
+~~~~~~~~~~~~~~
+
+Creating tasks follows the standard `celery
+conventions <http://docs.celeryproject.org/en/latest/userguide/tasks.html>`__.
+The only difference is the celery application that decorates the
+function should be imported from ``girder_worker.app``. E.g.:
+
+.. code:: python
+
+    from girder_worker.app import app
+
+    @app.task
+    def fibonacci(n):
+        if n == 1 or n == 2:
+            return 1
+        return fibonacci(n-1) + fibonacci(n-2)
+
+Plugin Class
+~~~~~~~~~~~~
+
+Each plugin must define a plugin class the inherits from
+``girder_worker.GirderWorkerPluginABC``. GirderWorkerPluginABC's
+interface is simple. The class must define an ``__init__`` function and
+a ``task_imports`` function. ``__init__`` takes the girder\_worker's
+celery application as its first argument. This allows the plugin to
+store a reference to the application, or change configurations of the
+application as necessary. The ``task_imports`` function takes no
+arguments and must return a list of the package paths (e.g. importable
```

Comment:
I have a few ideas how we might support this kind of functionality.  For now I'd like to keep this simple and look at implementing that in a different PR.